src/Quasar/Mount/Common/Gen.purs

```diff
@@ -21,50 +21,56 @@ module Quasar.Mount.Common.Gen
 import Control.Monad.Rec.Class (class MonadRec)
 import Data.Char.Gen as CG
-import Data.Either (Either(..))
+import Data.Maybe (Maybe(..))
 import Data.NonEmpty ((:|))
-import Data.Path.Pathy.Gen (genAbsDirPath, genAbsFilePath) as PGen
-import Data.String as S
 import Data.String.Gen as SG
-import Data.Tuple (Tuple(..))
-import Data.URI as URI
-import Quasar.Mount.Common (Credentials(..))
-import Quasar.Mount.MongoDB as MDB
-import Quasar.Types (AnyPath)
+import Data.String.NonEmpty (NonEmptyString)
+import Data.String.NonEmpty as NES
+import Data.These (These(..))
+import Pathy.Gen (genAbsDirPath, genAbsFilePath) as PGen
+import Quasar.URI as URI
+import URI.Host.Gen as HostGen
+import URI.Port as Port
 
 genAlphaNumericString ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m String
-genAlphaNumericString = SG.genString $ Gen.oneOf $ CG.genDigitChar :| [CG.genAlpha]
+genAlphaNumericString = SG.genString genAlphaNumericChar
 
-genHostURI ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m URI.Host
-genHostURI = Gen.oneOf $ genIPv4 :| [genName]
-  where
-  genIPv4 = do
-    a ← Gen.chooseInt 1 254
-    b ← Gen.chooseInt 1 254
-    c ← Gen.chooseInt 1 254
-    d ← Gen.chooseInt 1 254
-    pure $ URI.IPv4Address $ S.joinWith "." $ show <$> [a, b, c, d]
-  genName = do
-    head ← S.singleton <$> CG.genAlpha
-    tail ← genAlphaNumericString
-    pure $ URI.NameAddress $ head <> tail
+genAlphaNumericNEString ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m NonEmptyString
+genAlphaNumericNEString = NES.cons <$> genAlphaNumericChar <*> SG.genString genAlphaNumericChar
 
-genPort ∷ ∀ m. MonadGen m ⇒ m URI.Port
-genPort = URI.Port <$> Gen.chooseInt 50000 65535
+genAlphaNumericChar ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m Char
+genAlphaNumericChar = Gen.oneOf $ CG.genDigitChar :| [CG.genAlpha]
 
-genHost ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m MDB.Host
-genHost = Tuple <$> genHostURI <*> GenC.genMaybe genPort
+genPort ∷ ∀ m. MonadRec m ⇒ MonadGen m ⇒ m URI.Port
+genPort = filtered $ Port.fromInt <$> Gen.chooseInt 50000 65535
 
-genCredentials ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m Credentials
-genCredentials =
-  Credentials <$> ({ user: _, password: _ }
-    <$> genAlphaNumericString
-    <*> Gen.choose (pure "") genAlphaNumericString)
+genHost' ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m URI.QURIHost'
+genHost' = genThese HostGen.genHost genPort
+
+genHost ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m URI.QURIHost
+genHost = genMaybe genHost'
+
+genHosts ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m URI.QURIHosts
+genHosts = Gen.unfoldable genHost'
 
-genAnyPath ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m AnyPath
-genAnyPath = Gen.oneOf $ (Left <$> PGen.genAbsDirPath) :| [Right <$> PGen.genAbsFilePath]
+genThese ∷ ∀ m a b. MonadGen m ⇒ MonadRec m ⇒ m a → m b → m (These a b)
+genThese ma mb = filtered do
+  a' ← GenC.genMaybe ma
+  b' ← GenC.genMaybe mb
+  pure case a', b' of
+    Just a, Just b → Just $ Both a b
+    Just a, Nothing → Just $ This a
+    Nothing, Just b → Just $ That b
+    Nothing, Nothing → Nothing
```

Comment:
We can use the one of these I just added to `purescript-these` now :smile: (sorry)